### DISCUSSION TITLE
Option to specify file extension for VerifyZip and have it save as this file rather than the exploded zip

### DISCUSSION THREAD

**Discussion by MJB222398** [VerifyTests/Verify#1127] [Ideas]
Generally when verifying we have the option to specify the file extension and the verified file is saved with this. But for VerifyZip so far as I can see this option is not present because it saves the verified contents as the exploded zip structure. I would like to be able to pass a file extension into the method VerifyZip - typically would be '.zip', '.7z' etc. and have it actually save the verified file as a single compressed file with this extension. 

I feel this is necessary because verifying the compressed file just using Verify() is not helpful in pointing out differences in a readable fashion but using VerifyZip does not leave me with the finished article that I want. The zip file for me is an input to a process and in the case of a failure I might want to use the zip file and feed it into that process. Right now I would have to zip it all up again myself. I would also have to strip off the '.verified'  bits from each individual component filename (as these names matter). If it was saved as a zip file and the compressed component files kept their original names this would suit me well.

Is the above feasible with Verify? Or maybe I am using Verify wrong

**Comment by SimonCropp** [comment]
happy to consider a pull request
